v3/orderedmap.go

```diff
@@ -0,0 +1,175 @@
+package orderedmap
+
+import "iter"
+
+type OrderedMap[K comparable, V any] struct {
+	kv map[K]*Element[K, V]
+	ll list[K, V]
+}
+
+func NewOrderedMap[K comparable, V any]() *OrderedMap[K, V] {
+	return &OrderedMap[K, V]{
+		kv: make(map[K]*Element[K, V]),
+	}
+}
+
+func NewOrderedMapWithElements[K comparable, V any](els ...*Element[K, V]) *OrderedMap[K, V] {
```

Comment:
Can you also include `NewOrderedMapWithCapacity` from this recent PR: #48

Reply:
Rebased and added the changes from #48 into the first commit